### ISSUE TITLE
What's the difference between `accelerator.save_model` and `accelerator.save_state`

### ISSUE THREAD

**Issue by IcyFeather233** [huggingface/accelerate#2683]
### System Info

```Shell
None
```

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [ ] My own task or dataset (give details below)

### Reproduction

I noticed that in [save and load doc](https://huggingface.co/docs/accelerate/v0.29.3/en/basic_tutorials/migration?save=single+checkpoint#save-and-load), it metioned 2 ways:

## 1

save: 

```
accelerator.wait_for_everyone()
accelerator.save_model(model, save_directory)
```

load:

```
unwrapped_model = accelerator.unwrap_model(model)
path_to_checkpoint = os.path.join(save_directory,"pytorch_model.bin")
unwrapped_model.load_state_dict(torch.load(path_to_checkpoint))
```

## 2

`save_state()` and `load_state()`

### Expected behavior

I wonder what's the difference between these two use method

**Comment by muellerzr** [comment]
It mentions it pretty clear there, one is for mid-training, one is not. Is there more confusion in there or specific questions I can maybe answer?

Model: https://huggingface.co/docs/accelerate/v0.29.3/en/basic_tutorials/migration?save=single+checkpoint#model
State: https://huggingface.co/docs/accelerate/v0.29.3/en/basic_tutorials/migration?save=single+checkpoint#state

**Comment by IcyFeather233** [comment]
1. `accelerator.save_model` only save the model, and `accelerator.save_state` can save the current state of the model, optimizer, random generators, and learning rate schedulers, am I right?
2. I know when I go throught saving_steps, I can use `save_model` to save, but when I encounter a SIGINT signal accidentally, which function I need to use? Can I write like this?

```python
# call back function register
signal.signal(signal.SIGINT, self.process_exit_handler)

# accidentally save
def process_exit_handler(self, signal_received, frame) -> None:
    self.accelerator.wait_for_everyone()
    self.accelerator.save_state(output_dir=self.train_config.train_state_dir, safe_serialization=False)
    sys.exit(0)

# normally save
def save_model(self, suffix: Union[str, int]) -> None:
    unwrapped_model = self.accelerator.unwrap_model(self.model)
    unwrapped_model.save_pretrained(
        self.train_config.model_file.format(suffix),
        is_main_process=self.accelerator.is_main_process,
        save_function=self.accelerator.save
)

def train():
  ...
    if is_keep_training:
        self.accelerator.load_state(input_dir=train_config.train_state_dir)
        self.accelerator.register_for_checkpointing(lr_scheduler)
```

I am new to this and maybe a little confused. Thanks for your patience!

**Comment by diffunity** [comment]
I am still having trouble figuring out a good way to use this framework for training and evaluation.

So, from the links @muellerzr mentioned, I can gather the following information.

There are three ways that your model weights can be saved

1. `save_model` method

```python
accelerator.wait_for_everyone()
accelerator.save_model(model, save_directory)
```

2. `save_pretrained` method
```python
from transformers import AutoModel

unwrapped_model = accelerator.unwrap_model(model)
unwrapped_model.save_pretrained(
    "path/to/my_model_directory",
    is_main_process=accelerator.is_main_process,
    save_function=accelerator.save,
)

model = AutoModel.from_pretrained("path/to/my_model_directory")
```

3. `save_state` method
```python
accelerator.save_state()
```

`1` & `2` can be loaded using `unwrapped_model.load_state_dict(torch.load(path_to_checkpoint))`. `2` can be loaded using `AutoModel.from_pretrained()`. `3` can be loaded using `accelerator.load_state()`.

What would be a good way to manage the checkpoints when I want to save the model checkpoints during training and access them anytime in the future without having to fire up an `accelerator` object?
